synchronization/src/main/java/de/cyface/synchronization/HttpConnection.java

```diff
@@ -208,24 +213,30 @@ public HttpResponse post(@NonNull final HttpURLConnection connection, final @Non
                 buffer = new byte[bufferSize];
 
                 // Write the binaries to the OutputStream
-                int progress = 0;
+                int bytesWritten = 0;
                 int bytesRead;
                 bytesRead = bufferedFileInputStream.read(buffer, 0, bufferSize);
                 while (bytesRead > 0) {
                     outputStream.write(buffer, 0, bufferSize);
-                    outputStream.flush();
-                    progress += bytesRead; // Here progress is total uploaded bytes
-                    progressListener.updatedProgress((progress * 100.0f) / binarySize);
+                    bytesWritten += bytesRead; // Here progress is total uploaded bytes
+                    progressListener.updatedProgress((bytesWritten * 100.0f) / binarySize);
 
                     bytesAvailable = bufferedFileInputStream.available();
                     bufferSize = Math.min(bytesAvailable, maxBufferSize);
                     bytesRead = bufferedFileInputStream.read(buffer, 0, bufferSize);
                 }
+                bytesWrittenToOutputStream += bytesWritten;
 
                 // Write MultiPart Tail boundary
                 outputStream.write(TAIL.getBytes());
+                bytesWrittenToOutputStream += TAIL.getBytes().length;
+
+                // Ensure we only write the "registered" number of bytes to the stream MOV-693
+                outputStream.flush(); // This way we can identify exceptions thrown by flush easier
+                Validate.isTrue(bytesWrittenToOutputStream == fixedStreamLength, "bytesWrittenToOutputStream "
```

Comment:
This way I want to ensure the problem is not that the wrong number of bytes to be written to the stream is registered.

Reply:
It's currently also crashing, so no change here.